pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -436,20 +452,105 @@ func (w *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) (reco
+		}
+
+		nextNominatedWorkers, err := getNextNominatedWorkers(log, group)
+		if err != nil {
+			log.V(2).Error(err, "nominating next worker clusters failed")
+			return reconcile.Result{}, err
+		}
+
+		nominatedWorkers = append(group.local.Status.NominatedClusterNames, nextNominatedWorkers...)
+		w.roundStartTimes[key] = now
+
+		wlPatch := workload.BaseSSAWorkload(group.local)
+		wlPatch.Status.NominatedClusterNames = nominatedWorkers
+		if err = w.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(kueue.MultiKueueControllerName)); err != nil {
+			log.V(2).Error(err, "Failed to patch nominated clusters", "workloadName", group.local.Name)
+			return reconcile.Result{}, err
+		}
+	default:
+		log.V(3).Info("External nomination of the worker clusters", "dispatcherName", w.dispatcherName)
+		nominatedWorkers = group.local.Status.NominatedClusterNames
+	}
+
+	log.V(4).Info("Synchronize nominated workers", "nominatedWorkers", nominatedWorkers)
+
 	var errs []error
-	for rem, remWl := range group.remotes {
-		if remWl == nil {
-			clone := cloneForCreate(group.local, group.remoteClients[rem].origin)
-			err := group.remoteClients[rem].client.Create(ctx, clone)
-			if err != nil {
-				// just log the error for a single remote
-				log.V(2).Error(err, "creating remote object", "remote", rem)
-				errs = append(errs, err)
+	// Create or keep only nominated workers, remove others.
+	nominatedSet := sets.New(nominatedWorkers...)
+
+	for rem, remoteWl := range group.remotes {
+		if _, isNominated := nominatedSet[rem]; isNominated {
+			if remoteWl == nil {
+				clone := cloneForCreate(group.local, group.remoteClients[rem].origin)
+				if err := group.remoteClients[rem].client.Create(ctx, clone); err != nil {
+					log.V(2).Error(err, "creating remote object", "remote", rem)
+					errs = append(errs, err)
+				}
+			}
+		} else {
+			if remoteWl != nil {
+				if err := client.IgnoreNotFound(group.RemoveRemoteObjects(ctx, rem)); err != nil {
+					log.V(2).Error(err, "removing non-nominated remote object", "remote", rem)
+					errs = append(errs, err)
+				}
+				group.remotes[rem] = nil
 			}
 		}
 	}
-	return reconcile.Result{}, errors.Join(errs...)
+	return reconcile.Result{RequeueAfter: retryAfter}, errors.Join(errs...)
+}
+
+// getNextNominatedWorkers returns the next set of nominated workers for incremental dispatching.
+// It nominates up to 3 remotes that have not yet been nominated, in sorted order.
+func getNextNominatedWorkers(log logr.Logger, group *wlGroup) ([]string, error) {
+	alreadyNominated := make(map[string]struct{}, len(group.local.Status.NominatedClusterNames))
+	for _, remote := range group.local.Status.NominatedClusterNames {
+		alreadyNominated[remote] = struct{}{}
+	}
+
+	workers := make([]string, 0, len(group.remotes))
+	for remoteWorker := range group.remotes {
+		if _, found := alreadyNominated[remoteWorker]; !found {
+			workers = append(workers, remoteWorker)
+		}
+	}
+	sort.Strings(workers)
```

Comment:
```suggestion
	slices.Sort(workers)
```

The Sort function of slices lib is recommended, now.